src/exception.cr

```diff
@@ -2,22 +2,38 @@ require "callstack"
 
 CallStack.skip(__FILE__)
 
+# Descendants of class `Exception` are used to communicate between raise and rescue statements in begin ... end blocks.
+# `Exception` objects carry information about the exception – its type (the exception’s class name),
+# an optional descriptive string, and optional traceback information.
+# `Exception` subclasses may add additional information like `OpenSSL::Error#code`.
 class Exception
+  # Returns message of this exception.
   getter message : String?
+  # Returns the previous exception at the time this exception was raised.
+  # This is useful for wrapping exceptions and retaining the original exception information.
   getter cause : Exception?
   property callstack : CallStack?
 
   def initialize(@message : String? = nil, @cause : Exception? = nil)
   end
 
+  # Returns any backtrace associated with the exception.
+  # The backtrace is an array of strings, each containing “0xAddress Function at #{File Line Column}”‘.
+  #
+  # See also: `CallStack`.
   def backtrace
     self.backtrace?.not_nil!
   end
 
+  # Returns any backtrace associated with the exception if the call stack exists.
+  # The backtrace is an array of strings, each containing “0xAddress Function at #{File Line Column}”‘.
+  #
+  # See also: `CallStack`.
   def backtrace?
     @callstack.try &.printable_backtrace
   end
 
+  # Appends a `message` of this exception to the given *io*.
```

Comment:
I do think that this to_s should be documented because often users want inspect_with_backtrace instead.